client/domain/FeedbackParagraphObjectFactory.js

```diff
@@ -140,6 +140,20 @@ tie.factory('FeedbackParagraphObjectFactory', [
       return new FeedbackParagraph(PARAGRAPH_TYPE_SYNTAX_ERROR, error);
     };
 
+    /**
+     * Returns a FeedbackParagraph created from a json object
+     */
+    FeedbackParagraph.createFromJson = function(json) {
```

Comment:
JSON is actually a string encoding, but I think you're passing in an object/dict? If so, maybe call it fromDict() and also add a corresponding toDict() method.

Also: missing tests?